### PR TITLE
More documentation regarding dialog breaks

### PR DESCRIPTION
**Pull request by tobiasvl**
Includes documentation of a possible bug.

I'm not entirely sure of the naming in this PR, but it's 3 AM.

### CODE REVIEW

File: src/code/credits.asm
Comment (tobiasvl) [inline src/code/credits.asm:6681]: Should probably document this with "top" and "bottom"

**Comment by daid** [comment]
Most of it looks good, but I want to do some research on the bug as well to make sure we understand it proper.

**Comment by tobiasvl** [comment]
> Most of it looks good, but I want to do some research on the bug as well to make sure we understand it proper.

Yes, please do, it'd be good to have another set of eyes on it.

I can try to explain my findings a bit better, with examples, so other people can take a look more easily too.

You can compare (with the text debugger) how these dialogs behave:

- `Dialog007`: Even number of text lines, terminated prematurely - prints two lines at a time
- `Dialog008`: Odd number of text lines, terminated prematurely - prints the final line alone
- `Dialog021`: Even number of full text lines plus one `@` - prints two lines at a time
- `Dialog011`: Odd number of full text lines plus one `@` - prints two lines at a time, including one final empty line **<- the potential bug**

The former two cases are obviously trivial, as they don't require the buggy check at all, but they're useful to see how I think the latter two cases were supposed to be handled - specifically, that `Dialog008` prints two lines, and then just the one final line, scrolling up one single line at the end.

As I understand the code, it's meant to cover the two latter cases, to avoid an `@` on a line by itself printing an empty line. However, it only works in even-numbered dialogs (excluding the final `@`) like `Dialog021`.

If you rearrange the checks, so that checking if the next character is an `@` before you check if `$1F` characters have been printed, `Dialog011` will behave like `Dialog008`, which I assume is the intended behavior.

File: src/code/home/dialog.asm
Comment (kemenaran) [inline src/code/home/dialog.asm:704]: I think @Xkeeper0 started tagging potential bugs with `@bug` (as several decomp or disassembly projects do).

```suggestion
    ; @bug: This check should be done after the next two
```

Comment (tobiasvl) [reply src/code/home/dialog.asm:704]: Smart - I know a couple of other bugs that we should probably tag like that then.

**Review by kemenaran — APPROVED** [review]

**Comment by tobiasvl** [comment]
@daid Have you had time to look at it?

**Comment by kemenaran** [comment]
I'd say we can merge this – @daid except if you found something inexact in the PR?

**Comment by daid** [comment]
I think it is correct. Just merge it, and if there is some detail missed then we can always correct it later.

**Pull request merged**
